gprofiler/profilers/java.py

```diff
@@ -540,7 +541,13 @@ def _find_rw_exec_dir(self) -> str:
             if not is_owned_by_root(full_dir.parent):
                 continue  # the parent needs to be owned by root
 
-            mkdir_owned_root(full_dir)
+            try:
+                mkdir_owned_root(full_dir)
+            except OSError as e:
+                # dir is not r/w, try next one
+                if e.errno == errno.EROFS:
+                    continue
+                raise
```

Comment:
Exception-based programming is kind weird here, why not simply adding `if not is_rw_exec_dir(full_dir.parent): continue ` at the beginning of the iteration?

Reply:
Well, the exception-based programming here is because I don't want to "check" if the dir is ro, I want to try and create it, and fail otherwise. `is_rw_exec_dir` is written exception-based too, because it's simpler & more correct to just try and do it. "noexec" for example is not easily tested w/o just TRYING to execute and seeing if it fails